### DISCUSSION TITLE
Pyenv, tk, Fails to Launch Application

### DISCUSSION THREAD

**Discussion by MrDaisyBates** [pyenv/pyenv#2771] [Q&A]
Hey, guys. I hope you're well and I hope I chose the correct venue for this. I posted this over on Kohya, also.

I'm using Manjaro (Linux) and, to setup Kohya with the correct Python, I needed to use pyenv to change my local python version to an appropriate version. This has seemed to work correctly.

However, one thing that seems to be problematic is that while I have tk installed on my operating system, due to me using pyenv to shim and route my python to a different location, it seems like tk might just not be installed with the pyenv python.
![image](https://github.com/pyenv/pyenv/assets/143578183/f91ffd9a-0320-4e09-8e9b-921f91e1767e)
I did have another issue at the top (that I've not researched yet, but if the answer is simple, I'm happy to hear it, but I know that is Kohya related most likely and not required here).

What I'm specifically referring to is the stuff towards the bottom.

I'm also NOT very savvy with coding/programming type stuff, so feel free to talk to me like I'm 5 here ahahaha.

I think I understand the problem (the pyenv version of python not having tk). But I don't understand what the solution looks like. I'm assuming there's a healthy linux community here, so I'm hoping I'm okay asking this here. 

How do I get tk correctly configured with my localized pyenv python?

Thanks in advance!

Edit: I have posted this on the Stable Diffusion Reddit and the Kohya pages. Whichever leads me to a solution, I will edit the solution in here just in case any future internet traveler happens to stumble on it and need it. 
![image](https://github.com/pyenv/pyenv/assets/143578183/83399d8d-6de1-4bc8-917c-0ae2d80f014b)

I did get rid of the LD_LIBRARY_PATH issue by typing the suggested command up there, but the tk issue still persists. I say I fixed it. Whenever I open a new terminal window and try to launch the gui, the library path issue comes back. So, I must not be permanently fixing it.
![image](https://github.com/pyenv/pyenv/assets/143578183/69d185a0-f1be-459b-bc2b-785ecaefb609)
![image](https://github.com/pyenv/pyenv/assets/143578183/d34cbe78-49b1-4f65-893b-fed0c6820ea7)

A user on Reddit suggested my answer might lie here, but I don't understand it.
https://stackoverflow.com/questions/60469202/unable-to-install-tkinter-with-pyenv-pythons-on-macos

**Comment by MrDaisyBates** [comment]
I'm still looking to solve this issue. What I believe is happening is that the tk inside of the pyenv folder is not setup correctly to accommodate kohya. So, despite having tk installed system wide, for whatever reason, the one in my .pyenv folder isn't handling the job? I'm not savvy at this. At all. 

I've also taken this problem up on the Manjaro forums. I'm hoping to get some support there since I'm not sure how active you guys are here on the git. 
https://forum.manjaro.org/t/tk-not-configured-to-work-with-python-pyenv/147163

**Comment by native-api** [comment]
Check if you have all [Python build requirements](https://github.com/pyenv/pyenv/wiki#suggested-build-environment) installed. The TKinter module (Python's Tcl/Tk bindings) requires Tcl/Tk's development package to build.
